docs/source/Loaders.rst

```diff
@@ -80,3 +80,119 @@ test categories. With tags set, users can select a subset of the
 tests found by the test resolver (also known as test loader). For
 more information about the test tags, visit
 `<WritingTests.html#categorizing-tests>`__
+
+Test References
+===============
+
+A Test Reference is a string that can be resolved into (interpreted as)
+one or more tests by the Avocado Test Resolver.
+
+Each resolver (a.k.a. loader) can handle the test references
+differently. For example, External Loader will use the Test Reference as
+an argument for the external command, while the File Loader will expect
+a file path.
+
+If you don't specify the loader that you want to use, all of the
+available loaders will be used to resolve the provided test references.
+One by one, the test references will be resolved by the first loader
+able to create a test list out of that reference.
+
+Below you can find some extra details about the specific builtin Avocado
+loaders.
+
+For Loaders introduced to Avocado via plugins (VT, Robot, ...), please
```

Comment:
Those two last paragraphs merged together would looked better to my eyes (but it's really really minor)